kernel/hashlib.h

```diff
@@ -245,6 +239,11 @@ struct hash_cstr_ops {
 			h.hash32(*(a++));
 		return h;
 	}
+	[[nodiscard]] static inline Hasher hash(const char *a) {
```

Comment:
Why can't we use the macro here?

Reply:
instead of calling the polymorphic `h.eat(...)` we explicitly call the `hash_into` method of self

Reply:
as for why we do *that*, it's because I'm paranoid about pointer conversions

Reply:
Can't we use hash_into in the macro helper?